frontend/src/components/link.js

```diff
@@ -3,6 +3,16 @@ import { Link } from 'react-router';
 import shortUUID from '../helpers/uuid'
 
 export class NomadLink extends Component {
+
+    findNodeNameById(nodeId) {
+        let r = Object.keys(this.props.nodeList).filter(_ => { return this.props.nodeList[_]['ID'] === nodeId; })
```

Comment:
Just a nit, can we use 'const' instead of 'let' for immutable variables? Also, can we replace the underscore with a clear identifier like 'node'. That makes for more human-readable code.